code/game/objects/items/wayfinding.dm

```diff
@@ -86,33 +86,36 @@
 	if(istype(I, /obj/item/pinpointer/wayfinding))
 		var/obj/item/pinpointer/wayfinding/WP = I
 		to_chat(user, "<span class='notice'>You put \the [WP] in the return slot.</span>")
-		var/rfnd_amt
-		if((!WP.roundstart || WP.owner != user.real_name) && synth_acc.has_money(TRUE)) //can't recycle own pinpointer for money if not bought; given by a neutral quirk
-			if(synth_acc.has_money(refund_amt))
-				rfnd_amt = refund_amt
-			else
-				rfnd_amt = synth_acc.account_balance
-			synth_acc._adjust_money(-rfnd_amt)
-			var/obj/item/holochip/HC = new /obj/item/holochip(user.loc)
-			HC.credits = rfnd_amt
+		var/money = TRUE
+		if(synth_acc.has_money(refund_amt) && !WP.roundstart && WP.owner != user.real_name) //exploiters bring a friend
+			synth_acc._adjust_money(-refund_amt)
+			var/obj/item/holochip/HC = new /obj/item/holochip(loc)
+			HC.credits = refund_amt
 			HC.name = "[HC.credits] credit holochip"
 			if(istype(user, /mob/living/carbon/human))
 				var/mob/living/carbon/human/H = user
 				H.put_in_hands(HC)
 		else
-			var/crap = pick(subtypesof(/obj/effect/spawner/bundle/costume)) //harmless garbage some people may appreciate
-			new crap(user.loc)
+			money = FALSE
+			var/costume = pick(subtypesof(/obj/effect/spawner/bundle/costume)) //I don't think anyone wants this
+			new costume(user.loc)
 		qdel(WP)
-		set_expression("happy", 2 SECONDS)
-		say("<span class='robot'>Thank you for recycling, [user.first_name()]! Here is [rfnd_amt ? "[rfnd_amt] credits." : "a freshly synthesized costume!"]</span>")
+		set_expression("veryhappy", 2 SECONDS)
+		var/refund = "some credits."
+		if(!money)
```

Comment:
Will this code be affected by the cost being 0?

Reply:
No, this checks if it's giving a monetary refund or clothes.